parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -164,21 +164,7 @@ write_files:
   permissions: "0644"
   owner: root
   content: |
-    {
-      "live-restore": true,
-      "log-driver": "json-file",
-      "log-opts":  {
-         "max-size": "50m",
-         "max-file": "5"
-      }{{if IsNSeriesSKU .}}
-      ,"default-runtime": "nvidia",
-      "runtimes": {
-         "nvidia": {
-             "path": "/usr/bin/nvidia-container-runtime",
-             "runtimeArgs": []
-        }
-      }{{end}}
-    }
+{{IndentString (GetDockerConfig (IsNSeriesSKU .VMSize)) 4}}
```

Comment:
I just realize IsNSeriesSKU takes in agentpoolProfile, so it should probably be {{IndentString (GetDockerConfig (IsNSeriesSKU .)) 4}} note the dot after IsNSeriesSKU

Reply:
I changed the function in aks-engine to take the vmsize directly, because it caused issues between master and agent pool profiles: https://github.com/Azure/aks-engine/blob/5bef06b1f395c34fa258db31407a4119e6fed8a2/pkg/engine/template_generator.go#L576

So this is correct. However I'm removing all this logic based on Tony's feedback.